front/lib/api/assistant/agent.ts

```diff
@@ -358,32 +341,33 @@ export async function* runMultiActionsAgent(
             message: `Action ${a.sId} is missing a name`,
           },
         } satisfies AgentErrorEvent;
+        return;
       }
-      const name = a.name ?? legacySpec?.name;
-      if (!name) {
-        // Unreachable
-        throw new Error("Unreachable: no name and no legacy spec.");
+
+      const runner = getRunnerforActionConfiguration(a);
+      const legacySpecRes =
+        await runner.deprecatedBuildSpecificationForSingleActionAgent(auth);
+      if (legacySpecRes.isErr()) {
+        return legacySpecRes;
```

Comment:
We don't yield an error here?

Reply:
Keeping same behaviour as before, the parent I guess will take care of it

Reply:
Ok it doesn't you are right